src/main/kotlin/org/wa/rceditor/application/model/ContributorItem.kt

```diff
@@ -0,0 +1,36 @@
+package org.wa.rceditor.application.model
+
+import javafx.beans.property.ObjectProperty
+import tornadofx.*
+import java.util.*
+
+class ContributorItem(text: String) {
+    val id: UUID = UUID.randomUUID()
+
+    var text: String by property(text)
+    val textProperty = getProperty(ContributorItem::text)
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (other?.javaClass != javaClass) return false
+
+        other as ContributorItem
+
+        if (id != other.id) return false
+
+        return true
+    }
+
+    override fun hashCode(): Int {
+        return id.hashCode()
+    }
+
+    override fun toString(): String {
+        return text
+    }
+}
+
+class ContributorItemModel(property: ObjectProperty<ContributorItem>) :
```

Comment:
I am seeing a lot of model classes, but I'm not sure you need all of them. In this case for example, the core data of the app is the metadata for a resource container from the manifest (dublin core and project list, I think). So your data model is almost exclusively from kotlin-resource-container already, and you can just display and update that data from the UI.

Reply:
Here why I used ViewItemModels: https://stackoverflow.com/questions/52700365/binding-textfield-to-the-listview-item